x/ccv/provider/keeper/grpc_query.go

```diff
@@ -49,37 +50,66 @@ func (k Keeper) QueryConsumerChains(goCtx context.Context, req *types.QueryConsu
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	chains := []*types.Chain{}
-	for _, chainID := range k.GetAllRegisteredConsumerIds(ctx) {
-		c, err := k.GetConsumerChain(ctx, chainID)
+	lastCID, ok := k.GetConsumerId(ctx)
+	if !ok {
+		return &types.QueryConsumerChainsResponse{}, nil
+	}
+
+	consumerIds := []string{}
+	phase := ConsumerPhase(byte(req.Phase))
```

Comment:
Phase should be optional.

Desired behavior:
### User sends a request with phase
* respond with chains in that particulat phase

### user sends a request without phase:
* respond with all chains (up to the page limit, if applicable)